code/game/machinery/computer/security.dm

```diff
@@ -2,311 +2,192 @@
+		"rank" = "Please input new assignment:",
+		"fingerprint" = "Please input new fingerprint hash:",
+		"p_stat" = "Please select new physical status:",
+		"m_stat" = "Please select new mental status:",
+		// Security
+		"criminal" = "Please select new criminal status:",
+		"mi_crim" = "Please input new minor crimes:",
+		"mi_crim_d" = "Please summarize minor crimes:",
+		"ma_crim" = "Please input new major crimes:",
+		"ma_crim_d" = "Please summarize major crimes:",
+		"notes" = "Please input new important notes:",
+	)
+	field_edit_choices = list(
+		// General
+		"sex" = list("Male", "Female"),
+		"p_stat" = list("*Deceased*", "*SSD*", "Active", "Physically Unfit", "Disabled"),
+		"m_stat" = list("*Insane*", "*Unstable*", "*Watch*", "Stable"),
+		// Security
+		"criminal" = list(SEC_RECORD_STATUS_NONE, SEC_RECORD_STATUS_ARREST, SEC_RECORD_STATUS_EXECUTE, SEC_RECORD_STATUS_INCARCERATED, SEC_RECORD_STATUS_RELEASED, SEC_RECORD_STATUS_PAROLLED, SEC_RECORD_STATUS_DEMOTE, SEC_RECORD_STATUS_SEARCH, SEC_RECORD_STATUS_MONITOR),
+	)
+
 /obj/machinery/computer/secure_data/Destroy()
-	active1 = null
-	active2 = null
+	record_general = null
+	record_security = null
 	return ..()
 
 /obj/machinery/computer/secure_data/attackby(obj/item/O, mob/user, params)
-	if(istype(O, /obj/item/card/id) && !scan)
-		user.drop_item()
-		O.forceMove(src)
-		scan = O
-		ui_interact(user)
+	if(tgui_login_attackby(O, user))
 		return
 	return ..()
 
-//Someone needs to break down the dat += into chunks instead of long ass lines.
 /obj/machinery/computer/secure_data/attack_hand(mob/user)
 	if(..())
 		return
 	if(is_away_level(z))
 		to_chat(user, "<span class='danger'>Unable to establish a connection</span>: You're too far away from the station!")
 		return
 	add_fingerprint(user)
-	ui_interact(user)
+	tgui_interact(user)
 
-/obj/machinery/computer/secure_data/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/machinery/computer/secure_data/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
 	if(!ui)
-		ui = new(user, src, ui_key, "secure_data.tmpl", name, 800, 800)
+		ui = new(user, src, ui_key, "SecurityRecords", name, 800, 800)
 		ui.open()
+		ui.set_autoupdate(FALSE)
 
-/obj/machinery/computer/secure_data/ui_data(mob/user, ui_key = "main", datum/topic_state/state = GLOB.default_state)
+/obj/machinery/computer/secure_data/tgui_data(mob/user, ui_key = "main", datum/topic_state/state = GLOB.default_state)
 	var/data[0]
-	data["temp"] = temp
-	data["scan"] = scan ? scan.name : null
-	data["authenticated"] = authenticated
-	data["screen"] = screen
-	if(authenticated)
-		switch(screen)
+	data["currentPage"] = current_page
+	data["isPrinting"] = is_printing
+	tgui_login_data(data, user)
```

Comment:
Unsure how this handles aghosts.

Reply:
Aghosts weren't supported by the `login` plugin but now they are.

Admins will now see a **CentCom Secure Login** (or whatever you'd call it):
![image](https://user-images.githubusercontent.com/1496804/94076601-c280de00-fdfd-11ea-9a76-73138b83e4c5.png)

The login info will display the following:
![image](https://user-images.githubusercontent.com/1496804/94076622-ce6ca000-fdfd-11ea-82cb-f0a1a7048f74.png)

Note that on systems that check for the ID's access list (like you need to be Magistrate to set executions), Aghost logins are considered without access.

Reply:
And whenever an admin logs in through that button, a message to admins is sent:
![image](https://user-images.githubusercontent.com/1496804/94077106-1d1a3a00-fdfe-11ea-9109-489f94ae52d1.png)

Reply:
A few things:
- If you set someone to arrest, what shows in their records?
- It should support setting people to execute (their "ID" should be considered to have all_access() + get_all_centcom_access())
- Not needed to message all admins when any admin logs into any device.